### DISCUSSION TITLE
Inter-process communication over the network

### DISCUSSION THREAD

**Discussion by Jay1876** [nxrighthere/UnrealCLR#242] [Help]
I am new to unreal and have some knowledge of C# just researching and learning at this stage.

I was wondering if its possible to have one control app that drives multiple engines using the Unreal CLR. e.g scenarios control app sends "Hello world" from one machine and it updates 5 or more engines at once over IP. I asked this question because when setting up the Unreal CLR theres a close relationship between the project and the Unreal CLR? I had to execute dotnet run and it ask for a specific path.

**Comment by nxrighthere** [comment]
Yes, you can use this plugin for inter-process communication over the network, just like in a regular .NET application.

**Reply by Jay1876** [reply]
@nxrighthere  Have you got any sample/example application of inter-process communication with Unreal CLR even if its just printing hello world to multiple machine using 1 control software. And the most important question is how is this Unreal CLR communicating to Unreal Engine(e.g over TCP 127.0.0.1 Port 99999) ?

**Reply by nxrighthere** [reply]
Unfortunately, I don't have such code examples.

**Reply by Jay1876** [reply]
@nxrighthere What are you using for communication? how does data get to and from This Library to The Unreal project?

**Reply by nxrighthere** [reply]
I don't use inter-process communication personally, but I know that you at least can open a TCP socket and connect to the other end using it. Also, please, see https://github.com/nxrighthere/UnrealCLR/discussions/129
